src/goto-synthesizer/enumerative_loop_contracts_synthesizer.cpp

```diff
@@ -158,6 +159,23 @@ void enumerative_loop_contracts_synthesizert::init_candidates()
           get_assigns(
             local_may_alias, loop_head_and_content.second, assigns_map[new_id]);
 
+          // Don't check assignable for CPROVER symbol
+          for(auto it = assigns_map[new_id].begin();
+              it != assigns_map[new_id].end();) // no ++it
```

Comment:
Out of curiosity, why no `++it` here?

Reply:
That's done within the loop body: `erase` will invalidate the iterator.